packages/server/utils/MattermostServerManager.ts

```diff
@@ -0,0 +1,70 @@
+import fetch from 'node-fetch'
+
+// Mattermost is a server-only integration for now, unlike the Slack integration
+
+// We're following a similar manager pattern here should we wish to refactor the
+// Mattermost integration. We might want to do this if we decide to use Mattermost's
+// upcoming "App" framework
+
+export interface MattermostApiResponse {
+  ok: boolean
+  status: number
+  error: string | undefined
+}
+
+abstract class MattermostManager {
+  webhookUrl: string
+  abstract fetch
+  headers: any
+
+  constructor(webhookUrl: string) {
+    this.webhookUrl = webhookUrl
+  }
+
+  private async post(payload: any): Promise<MattermostApiResponse> {
+    const res = await fetch(this.webhookUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json'
+      },
+      body: JSON.stringify(payload)
+    })
+    return {
+      ok: res.ok,
+      status: res.status,
+      error: !res.ok ? (await res.json())?.message : undefined
+    }
+  }
+
+  async postMessage(textOrAttachmentsArray: string | unknown[], notificationText?: string) {
+    const prop =
+      typeof textOrAttachmentsArray === 'string'
+        ? 'text'
+        : Array.isArray(textOrAttachmentsArray)
+        ? 'attachments'
+        : null
+    if (!prop) throw new Error('Invalid matterMost message')
```

Comment:
nit: all lowercase mattermost? 

for github, the rule I use is it's either `GitHub` or `github`, never `gitHub`.